### PR TITLE
Added function to evaluate connection's decoding

### PR DESCRIPTION
**Pull request by hunse**
This required a minor change to the builder, ~~which may break derivative builders~~.

Also fixed a bug in `nengo.utils.ensemble.tuning_curves` to avoid input points being modified.

EDIT: I refactored this to keep the `build_linear_system` function, so that nothing should break in other builders.

### CODE REVIEW

File: nengo/utils/connection.py [hunk 1]
```diff
@@ -57,3 +57,49 @@ def function(x):
...
+    eval_points : ndarray (N, E)
+        A shallow copy of the evaluation points used. E is the dimensionality
+        of the connection input ensemble (i.e. `conn.size_in`).
+    targets : ndarray (N, D)
+        The target function value at each evaluation point.
+    decoded : ndarray (N, D)
+        The decoded function value at each evaluation point.
+    """
+    from nengo.builder.ensemble import get_activities
+    from nengo.builder.connection import get_targets
```
Comment (hunse) [inline nengo/utils/connection.py]: Something was breaking when I put these imports at the top of the file, so I put them here.

Comment (tbekolay) [inline nengo/utils/connection.py]: Circular imports, I'd guess.

Comment (hunse) [inline nengo/utils/connection.py]: Maybe, but the exception didn't read like that.

Comment (tbekolay) [inline nengo/utils/connection.py]: ¯\_(ツ)_/¯

**Comment by jgosmann** [comment]
I foresee merge conflicts with #677 ...

**Comment by hunse** [comment]
Likely, but I don't think they should be too hard to resolve. I didn't change too much here.

I'm more worried about all the other PRs on the stack right now. I forsee a lot of merge conflicts in the near future.

File: nengo/utils/connection.py (hunk 1 above)
Comment (tbekolay) [inline nengo/utils/connection.py]: Looked into it... I suppose it's not a circular import at the module level, but it is on the package level... this causes the `utils` package to depend on `builder`, but `builder` already depends on `utils` so neither can be imported first. So yeah, I think putting the import here is the right call.

**Comment by tbekolay** [comment]
This LGTM. Needs a changelog entry, and a test if it's not too much trouble.

**Comment by hunse** [comment]
Added a changelog entry and a test.

**Comment by tbekolay** [comment]
Made a few changes; give them a look @hunse and I'll merge if they look good to you.

As an aside, the Sigmoid neuron type gives interesting results for this!

utils test_connection test_eval_point_decoding

**Comment by hunse** [comment]
LGTM. You're impressed with the sigmoid because the errors are smaller and more regular than for e.g. LIF?

**Comment by tbekolay** [comment]
Not impressed, just interested that it's such a regular pattern. Nothing impresses me anymore...

**Pull request merged**